src/buildParameters.ts

```diff
@@ -71,100 +59,97 @@ export function buildParameters(uri: Uri): EvaluationParameters {
   return evaluationParams;
 }
 
-function getFhirVersion(): string | null {
-  const fhirVersionRegex = /using (FHIR|"FHIR") version '(\d(.|\d)*)'/;
-  const matches = window.activeTextEditor!.document.getText().match(fhirVersionRegex);
-  if (matches && matches.length > 2) {
-    const version = matches[2];
-    if (version.startsWith('2')) {
-      return 'DSTU2';
-    } else if (version.startsWith('3')) {
-      return 'DSTU3';
-    } else if (version.startsWith('4')) {
-      return 'R4';
-    } else if (version.startsWith('5')) {
-      return 'R5';
-    }
-  }
+function getCqlCommandArgs(
+  fhirVersion: string,
+  optionsPath: Uri,
+  libraryDirectory: Uri,
+  libraryName: string,
+  expression: string | undefined,
+  terminologyPath: Uri,
+  connection: Connection | undefined,
+  contexts: Map<string, Context>,
+  measurementPeriod: string,
+): string[] {
+  const args = ['cql'];
 
-  return null;
-}
+  args.push(`-fv=${fhirVersion}`);
+  if (optionsPath && fs.existsSync(optionsPath.fsPath)) {
+    args.push(`-op=${optionsPath}`);
+  }
+  const modelType = 'FHIR';
+  let modelPath: string | undefined = connection?.endpoint;
+
+  if (contexts) {
+    contexts.forEach((value: Context, key: string) => {
+      args.push(`-ln=${libraryName}`);
+      args.push(`-lu=${libraryDirectory}`);
+
+      if (expression && expression != undefined) {
+        args.push(`-e=${expression}`);
+      }
+
+      if (terminologyPath) {
+        args.push(`-t=${terminologyPath}`);
+      }
+
+      if (connection?.name === 'Local') {
+        // connection.endpoint = projectPath.toString(); Can't use projectPath because Evaluator is not ok with cql-option.json and other files that are not fhir resources.
+        modelPath = key;
+      }
+      if (modelPath) {
+        args.push(`-m=${modelType}`);
+        args.push(`-mu=${modelPath}`);
+      }
+
+      if (measurementPeriod && measurementPeriod !== '') {
+        args.push(`-p=${libraryName}."Measurement Period"`);
+        args.push(`-pv=${measurementPeriod}`);
+      }
+      args.push(`-c=${value.resourceType}`);
+      args.push(`-cv=${value.resourceID}`);
+    });
+  }
 
-interface TestCase {
-  name: string | null;
-  path: Uri | null;
+  return args;
 }
 
-/**
- * Get the test cases to execute
- * @param testPath the root path to look for test cases
- * @returns a list of test cases to execute
- */
-function getTestPaths(testPath: Uri, libraryName: string): TestCase[] {
+function getLocalContexts(testPath: Uri, libraryName: string): Map<string, Context> {
+  let testCases: Map<string, Context> = new Map<string, Context>();
   if (!fs.existsSync(testPath.fsPath)) {
-    return [];
+    return testCases;
   }
-
-  let testCases: TestCase[] = [];
   let directories = glob
     .sync(testPath.fsPath + `/**/${libraryName}`)
     .filter(d => fs.statSync(d).isDirectory());
   for (var dir of directories) {
```

Comment:
Please use `let` instead of `var` for loop variables.
```suggestion
  for (let dir of directories) {
```